### ISSUE TITLE
Remove the default push location when uploading/pushing packages

### ISSUE THREAD

**Issue by AdmiringWorm** [chocolatey/choco#2629]
The use of a default URL when pushing packages is deprecated and is currently scheduled for removal from Chocolatey in the future.

Instead, it is recommended to always push packages while specifying the source to use.

https://github.com/chocolatey/choco/blob/develop/src/chocolatey/infrastructure.app/commands/ChocolateyPushCommand.cs#L149-L163

**Comment by TheCakeIsNaOH** [comment]
This is likely to break the workflow a lot of people that are pushing to CCR. Currently, it looks like that there is not a warning that using `--source` will be required, only that note in the help output. 

Also, it would be good to have the default source config option available.

So perhaps the order of operations could be like this:
1. Push off removal to v2
2. Merge #2430 (from issue #63)
3. Add a deprecation warning on `choco push` if `--source` is not specified and `DefaultPushSource` is unset.
4. Remove the default push location in v2

**Comment by AdmiringWorm** [comment]
@TheCakeIsNaOH I completely agree with you, and the order of operations that you have mentioned yourself is more or less what we have agreed on doing it.

**Comment by vexx32** [comment]
I've filed #3077 to track the deprecation of this, which we'll need to have done before we can remove this entirely. #2430 has been merged and I'll submit a follow-up PR to add some deprecation messages and whatnot to sort out #3077.

Reassigning this to 3.0 milestone as we can't remove this in any 2.x version as it has not been deprecated in 1.x.
